tests/light/src/syslog_ng_config/statements/destinations/mqtt_destination.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2022 One Identity
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+import paho.mqtt.subscribe as subscribe
+
+from src.syslog_ng_config.statements.destinations.destination_driver import DestinationDriver
+
+
+class MQTTDestination(DestinationDriver):
+    def __init__(self, mqtt_config, **options):
```

Comment:
My idea is to separate syslog-ng related config options and Mqtt broker related config options. 
It would be good if we can follow an IO class separation pattern of this logic, as it  used in `FileIO()` or `NetworkIO()`
What do you think about that?

Reply:
I understand your point!
While I developed this feature, I felt this isn't the best solution (1 object response for 2 things - that break the SOLID). The reason to do that, is I don't want to make two separate objects to configure the MQTT. But I think it would be greater to make two separate objects. 
I'm trying to get any good idea to "defend" the current solutions, but I don't find any :( (probably because this isn't the best solution) 
I will check the idea that you described. As soon as I can, I will process every review comment